src/drake/fs.clj

```diff
@@ -16,6 +16,19 @@
 (def drake-ignore "Names of files or directories to be ignored by Drake"
   #{"_logs"})
 
+(defn get-directory
+  [path]
+  (loop [f (fs/file path)]
+    (cond
+     (.isDirectory f) f
+     (nil? f) nil
+     :else (recur (.getParentFile f)))))
+
```

Comment:
Why does `get-directory-path` exist at all? There's no value in turning a nice rich `File` object into an opaque `String` that we can't do anything interesting with. And indeed later on in `shell-helper` we just call `fs/file` on it again to convert it into a `File`.